File: Sirius/CodeTest/4_1D_array/no_5.py
'''// 5 4
// 1 2 3
// 3 4 4
// 1 4 1
// 2 2 2

// 1 2 1 1 0'''
lst=[]
idx,loop_time=map(int,input().split())
lst=[0 for _ in range(idx)]
while(loop_time>0):
    start_idx,end_idx,ball_num = map(int,input().split())
    lst[start_idx-1:end_idx] =[ball_num for _ in range(end_idx-start_idx+1)]
    loop_time-=1
for i in range(idx):
    print(lst[i],end=' ')

File: Sirius/CodeTest/4_1D_array/no_5.cpp
#include <iostream>
using namespace std;
int main(){
    int idx,loop_time,start_idx,end_idx,ball_num;
    cin >> idx >> loop_time;
    int bucket[idx]={0,};
    for(int i=0;i<loop_time;i++){
        cin >> start_idx >> end_idx >> ball_num;
        for(int j=start_idx-1;j<end_idx;j++){
            bucket[j]=ball_num;
        }
    }
    for(int i=0;i<idx;i++){
        cout << bucket[i] << ' ';
    }
    return 0;
}